### DISCUSSION TITLE
`dab start` errors don't provide a lot of guidance.

### DISCUSSION THREAD

**Discussion by JerryNixon** [Azure/data-api-builder#1461] [Ideas]
If you have, for example, an invalid action in `actions` the resulting error is:

> Could not initialize the engine with the runtime config file: dab-config.json 

For developers unfamiliar with DAB, this error gives little guidance to address the issue.

Should we consider better error messages to help developers resolve syntax errors?

**Comment by seantleonard** [comment]
Yes, this is needed and I want to take a look at that sooner rather than later. A big improvement would be to eliminate the swallowing of exceptions/messaging in `PerformOnConfigChangeAsync(IApplicationBuilder app)`: https://github.com/Azure/data-api-builder/blob/e810143c6bbe58d056181dda32b3703ea2ebb8ad/src/Service/Startup.cs#L627-L631

**Comment by abhishekkumams** [comment]
I agree the above error is thrown. But i also see the error for invalid action above the initialization error.
I used the config with an entity containing action as "write".

 I also see the error `fail: Azure.DataApiBuilder.Service.Startup[0]
      Unable to complete runtime initialization. Refer to exception for error details.
      System.ArgumentException: Requested value 'write' was not found.`

Can you share the complete log details? Just to make sure we are also getting the required errors.

But i agree and we will work on cleaning the logs.
